src/lib.rs

```diff
@@ -734,34 +827,71 @@ mod tests {
         assert!(pk2.verify_detached(&sig2, &data2));
     }
 
+    #[test]
+    fn signing_seed() {
+        let data1 = generate_random_bytes(50);
+        let data2 = generate_random_bytes(50);
+
+        let mut seed_bytes = [0; SEED_BYTES];
+        seed_bytes.copy_from_slice(&generate_random_bytes(SEED_BYTES));
+        let seed = Seed::from_bytes(seed_bytes);
+
+        let (pk1, sk1) = gen_sign_keypair_from_seed(&seed);
+        let (pk2, sk2) = gen_sign_keypair_from_seed(&seed);
+
+        assert_eq!(pk1, pk2);
+        assert_eq!(sk1, sk2);
+
+        let sig1 = sk1.sign_detached(&data1);
+        let sig2 = sk2.sign_detached(&data2);
+
+        assert!(pk1.verify_detached(&sig1, &data1));
+        assert!(pk2.verify_detached(&sig2, &data2));
```

Comment:
We normally also `assert` a failure for the "misuse" case for comprehensiveness.

Reply:
Added!